boards/nxp/mimxrt1180_evk/board.c

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2024 NXP
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/devicetree.h>
+#include <zephyr/platform/hooks.h>
+#include <soc.h>
+
+#if CONFIG_BOARD_EARLY_INIT_HOOK
+void board_early_init_hook(void)
+{
+#if defined(CONFIG_ETH_NXP_IMX_NETC) && (DT_CHILD_NUM_STATUS_OKAY(DT_NODELABEL(netc)) != 0)
+	/* RMII mode */
+	BLK_CTRL_WAKEUPMIX->NETC_LINK_CFG[4] = BLK_CTRL_WAKEUPMIX_NETC_LINK_CFG_MII_PROT(1);
```

Comment:
Just curious, is this board specific? Can this go inside soc.c?

Reply:
It's board specific.
The phy mode setting depends on what PHY used on board and the PHY connect to MAC.